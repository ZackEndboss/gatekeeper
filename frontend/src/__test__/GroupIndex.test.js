import React from 'react';
import renderer from 'react-test-renderer';
import {MockStore} from './mockStore';
import {Provider} from 'react-redux';
import ConnectedGroupIndex, {GroupIndex} from '../GroupIndex';
// import {Link} from 'react-router-dom';

jest.mock('react-router-dom/Link', () => 'Link');
jest.mock('react-router-dom/NavLink', () => 'NavLink');

describe('GroupIndex component', () => {
    it('render', () => {
        const component = renderer.create(<GroupIndex groups={['group1', 'group2']}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('load from state', () => {
    it('should load the state correctly', () => {
        const storeMock = MockStore({
            'gates': {
                'ftx': {},
                'fty': {}
            }
        });

        const subject = (
            <Provider store={storeMock}>
                <ConnectedGroupIndex/>
            </Provider>);
        const component = renderer.create(subject);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
