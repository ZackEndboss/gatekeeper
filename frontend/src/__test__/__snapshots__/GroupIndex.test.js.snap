// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupIndex component render 1`] = `
<div>
  <Toolbar />
  <EditServiceDialog />
  <Link
    activeClassName="active"
    to="/"
  >
    Home
  </Link>
  <Link
    to="/group/group1"
  >
    group1
  </Link>
  <Link
    to="/group/group2"
  >
    group2
  </Link>
</div>
`;

exports[`load from state should load the state correctly 1`] = `
<div>
  <Toolbar />
  <EditServiceDialog />
  <Link
    activeClassName="active"
    to="/"
  >
    Home
  </Link>
  <Link
    to="/group/ftx"
  >
    ftx
  </Link>
  <Link
    to="/group/fty"
  >
    fty
  </Link>
</div>
`;
